mlir/lib/Dialect/Rock/Transforms/AffixTuningParameters.cpp

```diff
@@ -240,16 +245,17 @@ deriveGemm1TuningParams(OpBuilder &builder, AttentionOp op,
       gemm0TuningParams.getOutputSwizzle(), gemm0TuningParams.getForceUnroll());
 }
 
-void AffixTuningParameters::affixTuningParametersImpl(AttentionOp op) {
+template <typename Op>
+static LogicalResult
+affixTuningParametersImplAttentionPerf(Op op, func::FuncOp funcOp,
+                                       Type elemTypeQ, Type elemTypeK,
+                                       Type elemTypeV) {
   OpBuilder builder(op.getContext());
-  Type elemTypeQ = cast<MemRefType>(op.getQueries().getType()).getElementType();
-  Type elemTypeK = cast<MemRefType>(op.getKeys().getType()).getElementType();
-  Type elemTypeV = cast<MemRefType>(op.getValues().getType()).getElementType();
   bool isAccel = rock::isAccel(op.getFeatures());
   if (!isAccel) {
-    op.emitError("Currently, attention op is only supported on GPUs "
-                 "with matrix accelerator extentions");
-    return signalPassFailure();
+    return op.emitError("Currently, attention/GEMM+elementwise+GEMM op is only "
```

Comment:
```suggestion
    return op.emitError("Currently, " + op->name() "  + "....
```
You can use op->name() for better error message.